main.tf

```diff
@@ -17,12 +17,12 @@ resource "aws_elasticache_replication_group" "redis" {
 }
 
 resource "aws_elasticache_parameter_group" "redis_parameter_group" {
-  name        = "tf-redis-${var.name}-${data.aws_vpc.vpc.tags["Name"]}"
+  name        = "tf-redis-${var.name}-${replace(format("%.100s", lower(replace("${data.aws_vpc.vpc.tags["Name"]}", "_", "-"))), "/\\s/", "-")}"
```

Comment:
I used to take into account such specific situation, when you can't trust user (for eg, when values come from CI server or from branch names of random developers). :)

![keep-calm-and-dont-trust-user-input](https://cloud.githubusercontent.com/assets/393243/26359388/a25da886-3fd5-11e7-8e92-fad66680b86a.png)